docs/misc/error.md

```diff
@@ -1,21 +1,128 @@
 # Errors
 
-When using ABS, you might bump into errors within your code:
-when the interpreter finds an error, it will give up trying
-to evaluate the script and will exit with status code `99`:
+When using ABS, you might bump into errors within your code. When the interpreter finds an error, it will give up trying to execute the script and will exit with status code `99`.
 
+Note that there are 2 phases of the interpreter: parser and evaluator.
+
+When the parser phase encounters a syntax error it will continue to process the rest of the file and report all of the syntax errors it finds. 
+
+However, the evaluator phase will quit immediately when it encounters an evaluation error. Thus, you may need to run the ABS interpreter multiple times to find all the run-time errors.
+
+When you are running ABS interactively in the Run, Eval, Print Loop (REPL) the location of the error can only be the current line you just entered.
+
+However, when you are running ABS over a script file, even a small one, locating errors requires more help from the interpreter. ABS now provides `[line:column]` positions as well as the error line itself following the error message. Be aware that the column position is approximate because the parser does not always accurately pin-point the location of the offending token.
```

Comment:
```suggestion
However, when you are running ABS over a script file, even a small one, locating errors requires more help from the interpreter. ABS provides `[line:column]` positions as well as the error line itself following the error message. Be aware that the column position is approximate because the parser does not always accurately pin-point the location of the offending token.
```

Reply:
I have another question, why is the colum an approximation?